libs/SalesforceSDK/src/com/salesforce/androidsdk/push/PushNotificationInterface.kt

```diff
@@ -42,4 +45,14 @@ interface PushNotificationInterface {
      * @param data the notification payload.
      */
     fun onPushMessageReceived(data: Map<String?, String?>?)
+
+    /**
+     * This method is optional, you may simply return null.
+     *
+     * Use this method to provide the correct FirebaseMessaging instance if the one
+     * you intend to use is not the default.
+     *
+     * @return the FirebaseMessaging instance to use.
+     */
+    fun supplyFirebaseMessaging(): FirebaseMessaging?
```

Comment:
I believe this is still necessary for some apps due issues in Firebase.  I did remove the method that would allow customers to supply their own `FirebaseApp` as that should not be necessary.